pkg/collector/corechecks/system/iostats_nix.go

```diff
@@ -50,11 +50,17 @@ func roundFloat(val float64) float64 {
 
 // Can't use values from math.MaxUint* because C size for longs changes on 32/64 bit machines
 const maxLong = int64(^uint(0) >> 1)
+const maxLong32 = math.MaxUint32
 
 // Compute the increment between two iostats values, taking into account they can overflow
-func incrementWithOverflow(currentValue, lastValue uint64, maxValue int64) int64 {
+func incrementWithOverflow(currentValue, lastValue uint64) int64 {
 	ret := int64(currentValue - lastValue)
 	if ret < 0 {
+		maxValue := maxLong
```

Comment:
We can simply use `math.MaxUint64` and `math.MaxUint32` directly, we no longer need the constants.

Reply:
For 32bit os some value might go over the 32bit limit, I'm thinking about `ReadBytes` and `WriteBytes` that are multiplied by SectorSize). WDYT ?

Reply:
I don't think the logic works on 32-bit systems for `ReadBytes` and `WriteBytes`, since I think in this case you'll want `maxValue` to be `int64(SectorSize*math.MaxUint32)`.

TBH I don't see a way to make it work for these 2 fields without explicitly setting their `maxValue` to that "custom" value. Trying to guess the `maxValue` won't work for these fields.

Reply:
I was talking about replacing `maxLong` by `math.MaxUint64` and `maxLong32` by `math.MaxUint32`, which shouldn't change the behavior. 

About `ReadBytes` and `WriteBytes`, if they overflow after multiplying, there isn't much we can do from here.

Reply:
@OlivierGaca is right, our values are offset by 1 when the base counter are wrapping.